### ISSUE TITLE
Using fig-subcap with pdf document causes latex errors

### ISSUE THREAD

**Issue by warnes** [quarto-dev/quarto-cli#7115]
### Bug description

If a chunk has both the chunck options `label` and `fig-subcap`, an extra `figure` environment is created inside the outer `figure` environment, resulting in `LaTeX Error: Not in outer par mode.` when processed to generate PDF.

Example below.

### Steps to reproduce

````
---
title: "`fig-subcap` test"
format: pdf
execute:
    echo: false
    warning: false
editor: source
---

This chunk is successfully processed:

```{r}
#| label: fig-success
#| fig-cap: 'Caption Here'
plot(1:10)
plot(1:10)
```


However, adding a sub-caption results in `LaTeX Error: Not in outer par mode.`.

```{r}
#| label: fig-failure
#| fig-cap: 'Caption Here'
#| fig-subcap: 
#|  - 'Sub-caption 1 Here'
#|  - 'Sub-caption 2 Here'
plot(1:10)
plot(1:10)
```

Removing either the `label` *or* `fig-subcap` chunk option resolves the issue.

````

The error arises because an extra nested `figure` environment is created:

```latex
\begin{figure}

{\centering 

\begin{figure}

{\centering \includegraphics{fig-subcap-test_files/figure-pdf/fig-failure-1.pdf}

}

\caption{Sub-caption 1 Here}

\end{figure}

}

\caption{\label{fig-failure}Caption Here}

\end{figure}
```

### Expected behavior

Rendering to PDF generates a figure with both a caption and subcaption

### Actual behavior

```
$ quarto render fig-subcap-test.qmd 
- The project is out-of-sync -- use `renv::status()` for details.


processing file: fig-subcap-test.qmd
1/5              
2/5 [fig-success]
3/5              
4/5 [fig-failure]
5/5              
output file: fig-subcap-test.knit.md

pandoc 
  to: latex
  output-file: fig-subcap-test.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  title: '`fig-subcap` test'
  editor: source
  
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999993 (TeX Live 2021) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages

compilation failed- error
LaTeX Error: Not in outer par mode.

See the L LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.218 
       

see /Users/warnes/src/Community_First/fig-subcap-test.log for more information.
```

### Your environment

- RStudio 2023.09.0+463 
- R version 4.3.1 (2023-06-16)
- MacOS Ventura 13.5.2 (22G91)

### Quarto check output

```bash
$ quarto check

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.433
      Path: /Applications/RStudio.app/Contents/Resources/app/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.10
      Path: /opt/local/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by mcanouil** [comment]
I can't reproduce using the latest development version but I can with the latest stable (1.3.450).
Note that you are not running on the latest stable version.

You might want (if you can) upgrade to the pre-release version to no longer have this issue.

**Comment by cscheid** [comment]
This has been fixed on 1.4. (Also, do note that if you repeat the exact same plot in R devices, knitr will not generate the second one. So you're actually only getting one plot for your subcap example. Change the second line to `plot(2:20)` to get what you want)
